internal/terraform/node_resource_plan_instance.go

```diff
@@ -417,6 +423,17 @@ func (n *NodePlannableResourceInstance) managedResourceExecute(ctx EvalContext)
 			checkRuleSeverity,
 		)
 		diags = diags.Append(checkDiags)
+
+		if refreshDeferred != nil {
```

Comment:
nit: this confused me initially as I didn't realise the other deferred branch only happens if `!skipPlanChanges` . Maybe just add a comment explaining why we do this separately.